### PR TITLE
AlmaLinux-8 Support (Basic)

### PR DESCRIPTION
**Pull request by sscmh**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe):  Resolving OS dependancies

## What is the current behavior?

- Installation of Check_MK-Server aborts with a message  "unsupported OS found"

## What is the new behavior?

- Installation of Check_MK-Server is working on AlmaLinux-8

## Other information
- Patches only made for Server-Role so far, not for Agent-Role, Q&A on customer-site is in status "work-in-progress".

### CODE REVIEW

File: roles/server/README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# tribe29.checkmk.server
+# tribe29.checkmk.server - added setup for AlmaLinux-8
```
Comment (robin-tribe29) [inline roles/server/README.md:1]: Please remove this and add a changelog fragment per CONTRIBUTING.md.

File: roles/server/tasks/main.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
 - name: "Preflight - Fail if target host is not stable for this Role."
   ansible.builtin.fail:
-    msg: "This role is not stable for the target operating system, {{ ansible_distribution }} {{ ansible_distribution_major_version }}."
+    msg: "This role is not stable for the target operating system, {{ ansible_distribution }}-{{ ansible_distribution_major_version }}."
```
Comment (robin-tribe29) [inline roles/server/tasks/main.yml:4]: Please remove the dash, this is a human-readable message.

File: roles/server/vars/RedHat.yml [hunk 3]
```diff
@@ -27,6 +27,10 @@ checkmk_server_prerequisites_per_distro:
     - firewalld
     - dnf-plugins-core
     - python3-libsemanage
+  AlmaLinux:
+    - firewalld
+    - dnf-plugins-core
+    - python3-libsemanage
```
Comment (robin-tribe29) [inline roles/server/vars/RedHat.yml:33]: Please test, if this is necessary, as the packages mentioned are same as under the `RedHat` key, which is the default.

**Review by robin-tribe29 — CHANGES_REQUESTED** [review]
Thanks for your contribution!
Please work through my comments, then we should be ready to merge, if the none of the tests fail.

**Comment by robin-tribe29** [comment]
Hi @sscmh, while we appreciate your contribution, I see you are building an additional role and I do not see currently how that benefits this collection. Please focus on the AlmaLinux support in the existing server role.
If you want to do additional things, you can of course build your own role, but that should not be part of this pull request.

**Comment by sscmh** [comment]
Hi,

yes i understand, i have accidentilly "pushed" my code to "main"

Sorry
Marc


Von: Robin Gierse ***@***.***>
Gesendet: Dienstag, 21. März 2023 08:17
An: tribe29/ansible-collection-tribe29.checkmk ***@***.***>
Cc: HERRMANN Marc ***@***.***>; Mention ***@***.***>
Betreff: Re: [tribe29/ansible-collection-tribe29.checkmk] AlmaLinux-8 Support (Basic) (PR #286)


Hi @sscmh<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fsscmh&data=05%7C01%7Cmarc.herrmann%40soprasteria.com%7C11fbf1094b8446b6b30d08db29dc3f9b%7C8b87af7d86474dc78df45f69a2011bb5%7C0%7C0%7C638149798156689287%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=MInePzeD%2FhN9KpODfcXqPi2%2FKGjIqAhsrYLvNJXdvf4%3D&reserved=0>, while we appreciate your contribution, I see you are building an additional role and I do not see currently how that benefits this collection. Please focus on the AlmaLinux support in the existing server role.
If you want to do additional things, you can of course build your own role, but that should not be part of this pull request.

-
Reply to this email directly, view it on GitHub<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Ftribe29%2Fansible-collection-tribe29.checkmk%2Fpull%2F286%23issuecomment-1477373684&data=05%7C01%7Cmarc.herrmann%40soprasteria.com%7C11fbf1094b8446b6b30d08db29dc3f9b%7C8b87af7d86474dc78df45f69a2011bb5%7C0%7C0%7C638149798156689287%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=%2FPRwttpvcPH%2FPI%2Fj3qDaRDKicC6yBzecPr6c5q21cdY%3D&reserved=0>, or unsubscribe<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAXYGZCFMG63NKPXS3STP45TW5FIWJANCNFSM6AAAAAAV2NF4VQ&data=05%7C01%7Cmarc.herrmann%40soprasteria.com%7C11fbf1094b8446b6b30d08db29dc3f9b%7C8b87af7d86474dc78df45f69a2011bb5%7C0%7C0%7C638149798156689287%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=%2FtPHJ1B0k3LSZeGDpYE1GWmU35kU%2FjuXWCf1%2Fdmlas0%3D&reserved=0>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>


C2 - Eingeschränkte Verwendung

Sopra Steria SE - Vorsitzender des Aufsichtsrates: Cyril Malargé - Vorstand: Christian Wrage (Sprecher), Frédéric Munch - Sitz der Gesellschaft: Hamburg - HRB 151350 Amtsgericht Hamburg - USt-ID-Nr.: DE118671351

**Comment by macener** [comment]
roles/server/vars/RedHat.yml also needs an new entry under 
```checkmk_server_prerequisites_per_distro:
AlmaLinux-8:
    - firewalld
    - dnf-plugins-core
    - python3-libsemanage 
  ```
Otherwise some tasks in the role will not run.

**Comment by robin-tribe29** [comment]
> roles/server/vars/RedHat.yml also needs an new entry under
> 
> ```
> AlmaLinux-8:
>     - firewalld
>     - dnf-plugins-core
>     - python3-libsemanage 
> ```
> 
> Otherwise some tasks in the role will not run.

@macener I have not tested this, but the values are the same as in "RedHat", which is the OS family and should work as a "fallback" or "default". Is that not the case?

File: roles/server/vars/RedHat.yml [hunk 4]
```diff
@@ -29,7 +29,7 @@ checkmk_server_prerequisites_per_distro:
     - python3-libsemanage
 
 checkmk_server_prerequisites: "{{ checkmk_server_prerequisites_per_distro[ansible_facts.distribution ~ '-' ~ ansible_facts.distribution_major_version]
-  | default( checkmk_server_prerequisites_per_distro[ansible_facts.distribution]) }}"
+  | default( checkmk_server_prerequisites_per_distro[ ansible_distribution_file_variety ]) }}"
```
[the remark this answers is not in the document]
Comment (sscmh) [reply roles/server/vars/RedHat.yml:32]: well, this change has a lot of background to explain:
1. ansible_distribution_file_variety  is as string of value 'RedHat" for the distribution which are are from Redhat itself or are derived from Redhat like Centos, Almalinux, RockLinux, and so on. So its basically saying , this is the Redhat-Universe, we are basically using "rpm's", etc.
2. i made the change to reduce "redundancy", as you mentioned in a PR-comment a few weeks before
3. if i see the next comment, the redudancy at some other places maybe reduced furthermore ....

**Comment by sscmh** [comment]
> > roles/server/vars/RedHat.yml also needs an new entry under
> > ```
> > AlmaLinux-8:
> >     - firewalld
> >     - dnf-plugins-core
> >     - python3-libsemanage 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Otherwise some tasks in the role will not run.
> 
> @macener I have not tested this, but the values are the same as in "RedHat", which is the OS family and should work as a "fallback" or "default". Is that not the case?



> > roles/server/vars/RedHat.yml also needs an new entry under
> > ```
> > AlmaLinux-8:
> >     - firewalld
> >     - dnf-plugins-core
> >     - python3-libsemanage 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Otherwise some tasks in the role will not run.
> 
> @macener I have not tested this, but the values are the same as in "RedHat", which is the OS family and should work as a "fallback" or "default". Is that not the case?

If i take a deeper look at this code :
checkmk_server_prerequisites_per_distro:
  RedHat:
    - firewalld
    - dnf-plugins-core
    - python3-libsemanage
  CentOS-7:
    - firewalld
    - dnf-plugins-core
    - libsemanage-python
  CentOS-8:
    - firewalld
    - dnf-plugins-core
    - python3-libsemanage
  RedHat-7:
    - firewalld
    - libsemanage-python
  RedHat-8:
    - firewalld
    - dnf-plugins-core
    - python3-libsemanage


Hi,

for minimal redundancy, i would suggest to make the following conceptual changes for OS recognition and mapping to "rpm-packages" :
1. RedHat/Centos/Almalinux/RockyLinux/etc. are recognized by "**ansible_distribution_file_variety**"
2. the differences in RPM's are derived from the OS-Level like 7,8 or 9 -> this map's the OS-Release 7 to **libsemanage-python** and the rest (8,9 and the future-Releases >9) to -> **python3-libsemanage** 
Of course this only work's correctly for a longer time, if its periodically tested on "**all**" RedHat distributions

File: roles/server/vars/RedHat.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (robin-tribe29) [reply roles/server/vars/RedHat.yml:32]: Okay, then this is fine with me.

**Comment by robin-tribe29** [comment]
The changes look weird, but that looks cosmetic.
I consider this PR ripe to merge, so I will do so in preparation of the next release.

**Pull request merged**